plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java

```diff
@@ -71,40 +69,36 @@
 
   private static final String CONSOLE_NAME = "App Engine Deploy";
 
-  private ProjectFromSelectionHelper helper;
-
-  public StandardDeployCommandHandler() {
-    this(new FacetedProjectHelper());
-  }
-
-  @VisibleForTesting
-  StandardDeployCommandHandler(FacetedProjectHelper facetedProjectHelper) {
-    this.helper = new ProjectFromSelectionHelper(facetedProjectHelper);
-  }
-
   @Override
   public Object execute(ExecutionEvent event) throws ExecutionException {
     try {
-      IProject project = helper.getAppEngineStandardProject(event);
-      if (project != null) {
-        if (!checkProjectErrors(project)) {
-          MessageDialog.openInformation(HandlerUtil.getActiveShell(event),
-                                        Messages.getString("build.error.dialog.title"),
-                                        Messages.getString("build.error.dialog.message"));
-          return null;
-        }
-
-        IGoogleLoginService loginService = ServiceUtils.getService(event, IGoogleLoginService.class);
-        DeployPreferencesDialog dialog =
-            new DeployPreferencesDialog(HandlerUtil.getActiveShell(event), project, loginService);
-        if (dialog.open() == Window.OK) {
-          launchDeployJob(project, dialog.getCredential());
-        }
+      IProject project = ProjectFromSelectionHelper.getProject(event);
+      if (project == null) {
+        throw new AssertionError("Deploy menu enabled for non-project resources.");
+      }
+      IFacetedProject facetedProject = ProjectFacetsManager.create(project);
+      if (facetedProject == null) {
+        throw new AssertionError("Deploy menu enabled for non-faceted projects.");
```

Comment:
this should be an exception not an error; Errors are for major VM problems only.